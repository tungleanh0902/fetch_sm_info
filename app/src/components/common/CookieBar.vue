<template>
  <div v-if="!session.cookiesAccepted">
    <Bar :show="show" :bar-type="'primary'" @close="onClose">
      <span class="hide-on-mobile"
        >This site uses cookies to help improve your experience.</span
      >
      By using Lunie, you accept our
      <router-link to="/terms" class="link">Terms of Service</router-link> and
      <router-link to="/privacy" class="link"> Privacy Policy</router-link>.
    </Bar>
  </div>
</template>

<script>
import { mapState } from "vuex"
import Bar from "common/Bar"
export default {
  name: `cookie-bar`,
  components: {
    Bar,
  },
  data: () => ({
    show: true,
  }),
  computed: {
    ...mapState([`session`]),
  },
  methods: {
    onClose: function () {
      this.$store.dispatch(`setAnalyticsCollection`, true)
      this.$store.dispatch(`setErrorCollection`, true)
      this.$store.dispatch(`storeLocalPreferences`)
    },
  },
}
</script>
