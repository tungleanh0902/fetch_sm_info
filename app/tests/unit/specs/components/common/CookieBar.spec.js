import { shallowMount } from "@vue/test-utils"
import CookieBar from "common/CookieBar"

describe(`CookieBar`, () => {
  let wrapper, dispatch, $store

  beforeEach(async () => {
    dispatch = jest.fn()
    $store = {
      commit: jest.fn(),
      state: {
        session: {
          cookiesAccepted: false,
        },
      },
      dispatch,
    }

    wrapper = shallowMount(CookieBar, {
      mocks: {
        $store,
      },
      stubs: [`router-link`],
    })
  })

  it(`shows the cookie bar`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`can triggers cookie acceptance on close`, () => {
    wrapper.vm.onClose()
    expect(dispatch).toHaveBeenCalledWith(`setAnalyticsCollection`, true)
    expect(dispatch).toHaveBeenCalledWith(`setErrorCollection`, true)
    expect(dispatch).toHaveBeenCalledWith(`storeLocalPreferences`)
  })
})
