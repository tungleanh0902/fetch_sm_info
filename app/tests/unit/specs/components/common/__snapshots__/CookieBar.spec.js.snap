// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieBar shows the cookie bar 1`] = `
<div>
  <bar-stub
    bartype="primary"
    link=""
    linkcaption=""
    show="true"
  >
    <span
      class="hide-on-mobile"
    >
      This site uses cookies to help improve your experience.
    </span>
    
    By using Lunie, you accept our
    
    <router-link-stub
      class="link"
      to="/terms"
    >
      Terms of Service
    </router-link-stub>
     and
    
    <router-link-stub
      class="link"
      to="/privacy"
    >
       Privacy Policy
    </router-link-stub>
    .
  
  </bar-stub>
</div>
`;
